lovely/deck_skins.toml

```diff
@@ -0,0 +1,119 @@
+[manifest]
+version = "1.0.0"
+dump_lua = true
+priority = 0
+
+#========================================================#
+# Choose any rank for custom deck and use provided atlas #
+#========================================================#
+[[patches]]
+[patches.regex]
+target = "functions/misc_functions.lua"
+pattern = '''if _front and _front.suit and \(_front.value == 'Jack' or _front.value == 'Queen' or _front.value == 'King'\) then([\s\S]*?)end([\s\S]*?)end([\s\S]*?)end'''
+position = "at"
+payload = '''
+if _front and _front.suit and G.SETTINGS.CUSTOM_DECK and G.SETTINGS.CUSTOM_DECK.Collabs then
+    local collab = G.SETTINGS.CUSTOM_DECK.Collabs[_front.suit]
+    if collab and collab ~= 'default' then
+        local deckSkin = SMODS.DeckSkins[collab]
+        if deckSkin then
+            local hasRank = false
+            for i = 1, #deckSkin.ranks do
+                if deckSkin.ranks[i] == _front.value then hasRank = true break end
+            end
+            if hasRank then
+                local atlas = G.ASSET_ATLAS[G.SETTINGS.colourblind_option and deckSkin.hc_atlas or deckSkin.lc_atlas]
+                if atlas then
+                    if  deckSkin.posStyle == 'collab' then
+                        return atlas, G.COLLABS.pos[_front.value]
+                    elseif deckSkin.posStyle == 'suit' then
+                        return atlas, { x = _front.pos.x, y = 0}
+                    elseif deckSkin.posStyle == 'deck' then
+                        return atlas, _front.pos
+                    end
+                end
+            end
+        end
+    end
+end
+'''
+match_indent = true
+
+#=======================#
+# Extend custom deck ui #
+#=======================#
+[[patches]]
+[patches.regex]
+target = "functions/UI_definitions.lua"
+pattern = '''local face_cards = CardArea\(([\s\S]*?)\)'''
+position = "at"
+payload = '''
+
+local rankCount = 0
```

Comment:
Steamodded wants to preserve the behavior of the game for aspects that other mods don't actively change. As such, I can't approve always putting all ranks on display. At the very least the vanilla collabs should only display the face cards they're changing. Maybe you could make use of the `ranks` table to respect that?

Reply:
Only ranks with skins will be shown now. This occurs though all skins, so a '3' of Spades texture on 'MySkin' will cause a 3 to show up in all other Spade skins even without a 3. The vanilla deck skin tabs do not update the ranks they display between skins, only suits which causes this. I could force an update, but with the way the 'default' skin matches the collab skin ranks I believe this is the intent and fully shows the differences beetween skins.